lldb/source/Target/Process.cpp

```diff
@@ -3264,6 +3266,11 @@ Status Process::PrivateResume() {
   // filters before resuming.
   UpdateAutomaticSignalFiltering();
 
+  if (m_last_run_direction != direction) {
```

Comment:
It might be hard, but it would be nice if this wasn't necessary.  For instance, I do a forward-next over a function and hit a breakpoint.  In lldb at present, I can do a couple of steps around where I hit the breakpoint, then I can just "continue" to conclude the forward-next.  But with this change, if I hit the breakpoint, forward step, decide I stepped too far, backwards step, find what I wanted, and then do "continue" now I get an actual continue, we've forgotten about the forward-next.

Reply:
TBH I don't understand thread plans very well yet so I was trying to avoid interacting with them. I guess implementing your suggestion means extending thread plans with a `RunDirection`, and/or creating new plan subclasses? And then we'd have to test a lot of interactions. Would it be OK to defer that to a future patch to keep this patch reasonably small?

Reply:
I'm a bit worried that in general thread plans have a lot of complexity, and adding more complexity by allowing multiple stacked plans with different directions might be bad enough for maintainability that it's overall a bad idea.

Reply:
If the user does "continue", hits a breakpoint, does "reverse-step", and then "reverse-continue" presumably we do have to abandon the original "continue". Since this patch only implements "reverse-continue" maybe always abandoning the thread plans actually the right thing to do for now?